common/lib/xmodule/xmodule/partitions/tests/test_partitions.py

```diff
@@ -325,9 +325,19 @@ def test_get_user_group_id_for_partition(self):
 
         # switch to the second group and verify that it is returned for the user
         self.user_partition.scheme.current_group = groups[1]    # pylint: disable=no-member
-        group2_id = self.partition_service.get_user_group_id_for_partition(user_partition_id)
+        #group2_id = self.partition_service.get_user_group_id_for_partition(user_partition_id)
+        #self.assertEqual(group2_id, groups[1].id)    # pylint: disable=no-member
+
+        partition_service2 = StaticPartitionService(
+            [self.user_partition],
+            user=Mock(username='ma', email='ma@edx.org', is_staff=False, is_active=True),
+            course_id=self.course.id,
+            track_function=Mock()
+        )
+        group2_id = partition_service2.get_user_group_id_for_partition(user_partition_id)
```

Comment:
PartitionSchemes (at least, "real" ones) are read-only adapter interfaces to something else.  That is, there's nothing saying that data used by a specific partition scheme couldn't change during any request or other arbitrary lifecycle - the partition scheme doesn't make any assumptions about it, and the partition service just forwards what the scheme says.

Which is why I think a better implementation, if possible, would be to disable caching unless a request cache is passed in.  The passing in of a cache container would indicate that the caller intends to consider the underlying data static/immutable throughout the lifetime of the service, which seems like the behavior you might want during a grading run.  Does that make sense?